components/remote_settings/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+pub mod error;
+use config::RemoteSettingsConfig;
+pub use error::{RemoteSettingsError, Result};
+mod config;
+use rs_client::{Client, ClientConfig};
+use std::{fs::File, io::prelude::Write, sync::Mutex};
+
+uniffi_macros::include_scaffolding!("remotesettings");
+
+pub struct RemoteSettings {
+    pub config: RemoteSettingsConfig,
+    client: Mutex<Client>,
+}
+
+impl RemoteSettings {
+    pub fn new(config: RemoteSettingsConfig) -> Result<Self> {
+        Ok(RemoteSettings {
+            config: config.clone(),
+            client: Mutex::new(Client::new(config.into())?),
+        })
+    }
+
+    pub fn get_records(&self) -> Result<Vec<JsonObject>> {
```

Comment:
Each collection in Remote Settings can define an arbitrary schema for their records, so if I'm understanding this correctly I don't think it would work. Each call to the records endpoint will return on array of (record) objects that matches the arbitrary schema. Here's another example:

```
{
  "data": [
    {
      "field1": "a",
      "another-field": true,
      "string-type-2": "a",
      "int-type": 100,
      "schema": 1678231127074,
      "id": "c92e34b3-40e1-4989-86bb-c7f48eacab31",
      "last_modified": 1678231137883
    }
  ]
}
```